rokka/sourceimages.go

```diff
@@ -232,3 +232,64 @@ func (c *Client) DeleteDynamicMetadata(org, hash, name string, options DynamicMe
 	err = c.Call(req, &result, dynamicMetadataResponseHandler)
 	return result, err
 }
+
+// SetUserMetadata updates a source image by adding arbitrary metadata. If there were previous user metadata set on this source image,
+// they'll get overwritten.
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) SetUserMetadata(org, hash string, data io.Reader) error {
+	req, err := c.NewRequest(http.MethodPut, fmt.Sprintf("/sourceimages/%s/%s/meta/user", org, hash), data, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
+
+// UpdateUserMetadata updates a source image by adding arbitrary metadata. Previous values get merged with the new values.
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) UpdateUserMetadata(org, hash string, data io.Reader) error {
+	req, err := c.NewRequest(http.MethodPatch, fmt.Sprintf("/sourceimages/%s/%s/meta/user", org, hash), data, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
+
+// DeleteUserMetadata updates a source image by deleting existing metadata.
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) DeleteUserMetadata(org, hash string) error {
+	req, err := c.NewRequest(http.MethodDelete, fmt.Sprintf("/sourceimages/%s/%s/meta/user", org, hash), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
+
+// UpdateUserMetadataByName updates a source image by setting user metadata identified by name.
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) UpdateUserMetadataByName(org, hash, name string, data io.Reader) error {
+	req, err := c.NewRequest(http.MethodPut, fmt.Sprintf("/sourceimages/%s/%s/meta/user/%s", org, hash, name), data, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
+
+// DeleteUserMetadataByName updates a source image by removing user metadata identified by name.
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) DeleteUserMetadataByName(org, hash, name string) error {
+	req, err := c.NewRequest(http.MethodDelete, fmt.Sprintf("/sourceimages/%s/%s/meta/user/%s", org, hash, name), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
```

Comment:
Same goes for the UpdateUserMetadataByName and DeleteUserMetadataByName functions. Maybe we should create a helper function which gets called by these 2 functions with the right parameters in order to reduce redundancy?

Reply:
fixed, too